### DISCUSSION TITLE
Design Matrix Constraining Parameters

### DISCUSSION THREAD

**Discussion by dantzert** [bmcclintock/momentuHMM#65] [Q&A] [answered]
Hi Brett,

I'm trying to prescribe some expected relationships between behavioral states as in lines 31 - 60 of the [fur seal example](https://github.com/bmcclintock/momentuHMM/blob/master/vignettes/examples/nfsExample.R)

I'm having a hard time understanding the structure of the design matrices and how to change them to indicate the relationships I wish to. I'm thinking that the suffix numbers on the means and standard deviations refer to the stateNames list, but I'm confused as to the meaning of "(intercept)" and the semantic implication of 1's and 0's within the matrix. I'm presuming this is also a strict binary (i.e. no fractional values), is this correct?

Could you help me understand how to read / write matrices such as angleDM and stepDM in the example?

Thank you much,

Travis

**Comment by bmcclintock** [comment, marked as answer]
The pseudo-design matrix can have a steep learning curve, but typically once one has that "a-ha!" moment constructing them becomes straightforward. Unless you want to delve into statistical literature (e.g. https://en.wikipedia.org/wiki/Design_matrix), I would suggest starting out by carefully going through the entire momentuHMM [vignette](https://cran.r-project.org/web/packages/momentuHMM/vignettes/momentuHMM.pdf) and all of the examples, particularly Section 2 and the harbour seal example (Section 3.7). 

The pseudo-design matrix rows correspond to the real scale parameters and the columns correspond to the working scale parameters. Values can be binary, continuous, factors, or even characters (e.g. for individual-level covariates or 'as is' formula expressions). The capture-recapture software [Program MARK](http://www.phidot.org/software/mark/index.html) uses a similar design matrix framework for modeling real scale data stream distribution parameters (e.g., detection probability) and state transition probabilities (e.g. apparent survival probability) as a function of covariates that was largely the motivation  for `DM` specification in `momentuHMM`. To see how classic capture-recapture models are related to HMMs see, e.g., [Pradel 2005](https://onlinelibrary.wiley.com/doi/10.1111/j.1541-0420.2005.00318.x) and [McClintock et al. 2020](https://arxiv.org/abs/2002.10497). 

A good introduction to design matrix concepts can be found in [Chapter 6](http://www.phidot.org/software/mark/docs/book/pdf/chap6.pdf) of [Program MARK: A Gentle Introduction](http://www.phidot.org/software/mark/docs/book/). The main departure in `momentuHMM` is that you specify a separate pseudo-design matrix for the real scale parameters of each data stream (e.g. step length, turn angle, number of foraging dives). The specification of `DM` is very flexible but as a result is a more advanced topic; hopefully these resources will help you achieve that "a-ha!" moment.

**Comment by dantzert** [comment]
Thanks! Looking into those now.
